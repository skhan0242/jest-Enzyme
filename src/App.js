import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      count : 0
    }
  }

  makeIncrement = amount =>() =>{
    this.setState(prevState =>{
      return{
        count : prevState.count + amount
      }
    })
  }

  increment = this.makeIncrement(1);
  decrement = this.makeIncrement(-1);

  // increment = () =>{
  //   this.setState(prevState =>{
  //     return{
  //       count : prevState.count +1
  //     }
  //   })
  // }

  // decrement = () =>{
  //   this.setState(prevState =>{
  //     return{
  //       count: prevState.count-1
  //     }
  //   })
  // }
  render() {
    const { count } = this.state;
    return (
      <div className="App">
        <p>current count {count}</p>
        <button className="incrementBtn" onClick={this.increment}>+++</button>
        <button className="decrementBtn" onClick={this.decrement}>---</button>    
      </div>
    );
  }
}

export default App;
