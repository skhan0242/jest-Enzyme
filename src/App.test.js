import React from "react";

import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16"
import renderer from "react-test-renderer"
import App from "./App";

Enzyme.configure({adapter: new Adapter()})

describe('App Component',() =>{
  it('Matches the snapshot', () =>{
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('counter Text', () =>{
    const wrapper = shallow(<App />);
    const text = wrapper.find("p").text();
    expect(text).toBe('current count 0');
  });

  it('initail count should be 0', () =>{
    const wrapper = shallow(<App />);
    const count  = wrapper.state().count;
    expect(count).toEqual(0);
  });

  it('after increament it should increase one', () =>{
    const wrapper = shallow(<App />);
    const incrementButton = wrapper.find('button.incrementBtn');
    incrementButton.simulate('click');
    const incrementText = wrapper.find('p').text();
    expect(incrementText).toEqual("current count 1");
  });

  it('after decreament it should decrease one',() =>{
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find('.decrementBtn');
    decrementBtn.simulate('click');
    const decrementText = wrapper.find('p').text();
    expect(decrementText).toBe('current count -1')
  })
})