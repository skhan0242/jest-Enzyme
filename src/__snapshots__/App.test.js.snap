// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Matches the snapshot 1`] = `
<div
  className="App"
>
  <p>
    current count 
    0
  </p>
  <button
    className="incrementBtn"
    onClick={[Function]}
  >
    +++
  </button>
  <button
    className="decrementBtn"
    onClick={[Function]}
  >
    ---
  </button>
</div>
`;
